src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -61,22 +61,40 @@ public void storeMessage(Message message) {
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
-    for (Entity entity : results.asIterable()) {
+    createMessage(results, messages);
+
+    return messages;
+  }
+
+  public List<Message> getAllMessages() {
+    List<Message> messages = new ArrayList<>();
+
+    Query query = new Query("Message")
+      .addSort("timestamp",SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    createMessage(results, messages);
+
+    return messages;
+  }
+
+  private void createMessage(PreparedQuery pq, List<Message> m) {
```

Comment:
Also, small nit: method names and arguments serve as documentation and it's important that they accurately describe what they do. A name like `createMessage()` suggests that only one message is being created, so pluralizing may be more appropriate.

In general, try to avoid single-character variable names like `m`, as there's usually a more descriptive name (i.e. `messages`). Of course, there are exceptions to this, such as if you're using `x` and `y` as variables for coordinates in a grid.